### DISCUSSION TITLE
Better than staking: Method for stabilizing dogecoin price opt-in coinbase maturity extension

### DISCUSSION THREAD

**Discussion by GiverofMemory** [dogecoin/dogecoin#3007] [General]
@patricklodder you wanted a way to stake dogecoin without creating a new coin correct?  Well here is one solution to that.  Albiet it only applies to miners, however I will explain why that is best towards the end of this post.  The downside is we will have to increase emission by a small (insignificant and diminishing) predictble fraction.

This method wouldn't work with bitcoin since that is a limited asset and the more robust trading activity, the more stable the price.

However for dogecoin with constant emission, elongating the coinbase maturity (time it takes for your mined dogecoin to be ready to spend) the more stable the price.

So here is how it would work.  The miner puts a flag somewhere in the block they are mining that selects an elongated coinbase maturity.  Instead of the coins bieng ready to spend in 100 mins or whatever the stock setting is, they could opt for something like 1 year.  As a reward for doing this, they get an extra 4% mining reward.  This extra reward percentage would float and be based on current dogecoin inflation rate.  So when dogecoin inflation rate is 3%, then the reward for elongating your coinbase reward by 1 yr is 3%.  So basically in this case the miner would get to hold the coin for a year without risking losses due to inflation.  It isn't the most snazzy reward, but for miners that want to hold a percentage of their block rewards for the long term, it would make sense.  Perhaps there can be a choice, 1 yr, 2 yr, 5 yr, 10 yr and you get proportionally more block reward accordingly.  Think of it kind of like a mining bond.

Why is this better than letting anyone stake?  The reason is firstly because it is much more easily possible without jumping through magical (hackable) hoops.  Secondly because every block must be staked as it is mined, so there will never be a huge sum of expiring staked rewards.  If anyone could stake at any time and any amount then someone could stake a billion dollars worth and everyone would worry when that staked amount would expire.  This method does not have that problem in the slightest as each staked block would be able to be spent exactly 1 year after it was mined.  So there is no risk of a big dump.

This method should turn dogecoin into an unhackable stabilized coin (unlike hackable stablecoins).

**Comment by FesterCluck** [comment]
I'm trying to find an argument against this, but I can't. The features to support this seem available since 2021. It's not forced on the miner, and it'd only be the amount of the reward. It's essentially just extending the reward transaction to be a future spend. 

@GiverofMemory Regarding what I'll call the "Reward Interest" floating based on inflation, is the intent here that it would be based on a) the inflation rate at the time of mine or b) at the time of future spend? Also, could you give your reasoning behind the choice? As I understand it, I'd be in favor of a).
